src/test/java/com/vinylteam/vinyl/util/impl/JunoVinylParserTest.java

```diff
@@ -0,0 +1,97 @@
+package com.vinylteam.vinyl.util.impl;
+
+import com.vinylteam.vinyl.entity.Currency;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JunoVinylParserTest {
+
+    private final JunoVinylParser parser = new JunoVinylParser();
+
+    @Test
+    void getPageLinks() {
+        var pageLinks = parser.getPresentPageLinks();
+        assertFalse(pageLinks.isEmpty());
+    }
+
+    @Test
+    public void countPageLinksTest() {
+        Set<String> links = Set.of("https://www.juno.co.uk/all/back-cat/3/?media_type=vinyl", "https://www.juno.co.uk/all/back-cat/333/?media_type=vinyl");
+        Integer number = parser.countPageLinks(links);
+        assertEquals(333, number);
+    }
+
+    @Test
+    void givenLinksToOnePage_whenRawOffersAreReceived_thenCorrect() {
+        var pageLink = "https://www.juno.co.uk/all/back-cat/2/?media_type=vinyl";
+        var items = parser.readVinylsDataFromAllPages(Set.of(pageLink));
+        assertFalse(items.isEmpty());
+    }
+
+    @Test
+    void givenLinksToPages_whenFullListFromOneToMAxPageISReceived_thenCorrect() {
+        var items = parser.getFullPageLinksList(Set.of("https://www.juno.co.uk/all/back-cat/3/?media_type=vinyl", "https://www.juno.co.uk/all/back-cat/12/?media_type=vinyl"));
+        assertEquals(12, items.size());
+    }
+
+    @Test
+    void givenLinksToPages_whenRawOffersAreReceived_thenCorrect() {
+        var items = parser.readVinylsDataFromAllPages(Set.of("https://www.juno.co.uk/all/back-cat/3/?media_type=vinyl", "https://www.juno.co.uk/all/back-cat/12/?media_type=vinyl"));
+        assertFalse(items.isEmpty());
+    }
+
+    @Test
+    public void givenHtmlImgElement_whenRawOfferISCreated_thenCorrect() throws IOException {
```

Comment:
Why in  the test names you use CamelCase with under lines? Please use only CamelCase. 
Please add @DisplayName annotation on the top of each test to describe what exactly test doing